packages/mui-material-next/src/FormControl/formControlState.js

```diff
@@ -1,4 +1,6 @@
+// TODO v6: decide whether to update/refactor this, keep as-is, or drop it
```

Comment:
What's the idea behind dropping/refactoring it?

Reply:
This is always used together with `useFormControl` like:

```js
const muiFormControl = useFormControl()

const fcs = formControlState({
    props,
    muiFormControl,
    states: ['color', 'disabled'],
});
```

I was thinking it could be more intuitive to do this in a single hook, e.g.

```js
const [
  formControlState, // the result of `formControlState()`
  formControlContext, // the result of `useFormControl()`
] = useFormControlState({
  props,
  states: ['color', 'disabled'],
})
```

This would be internal, and wouldn't affect the original (public) [`useFormControl`](https://mui.com/material-ui/react-text-field/#useformcontrol) unless we wanted to (e.g. rename it to `useFormControlContext`)

Reply:
I see. Adding a higher-level abstraction (`useFormControlState`) makes sense to me if we're using `useFormControl` and `formControlState` together all the time 👍🏼 

We could create an issue and add it to the v6 milestone so we don't forget and can discuss the specifics whenever we have time.